entity-framework/core/modeling/generated-properties.md

```diff
@@ -47,48 +47,73 @@ Like `value generated on add`, if you specify a value for the property on a newl
 >
 > [!code-sql[Main](../../../samples/core/Modeling/FluentAPI/ValueGeneratedOnAddOrUpdate.sql)]
 
-## Conventions
+## Value generated on add
+
+By convention, non-composite primary keys of type short, int, long, or Guid are be set up to have values generated on add, and your database provider typically takes care of the necessary configuration. For example, an numeric primary key in SQL Server is automatically set up to be an IDENTITY column.
 
-By default, non-composite primary keys of type short, int, long, or Guid will be setup to have values generated on add. All other properties will be setup with no value generation.
+You can configure any property to be value generated on add as follows:
 
-## Data Annotations
+### [Data Annotations](#tab/data-annotations)
 
-### No value generation (Data Annotations)
+[!code-csharp[Main](../../../samples/core/Modeling/DataAnnotations/ValueGeneratedOnAdd.cs?name=ValueGeneratedOnAdd&highlight=5)]
 
-[!code-csharp[Main](../../../samples/core/Modeling/DataAnnotations/ValueGeneratedNever.cs#Sample)]
+### [Fluent API](#tab/fluent-api)
 
-### Value generated on add (Data Annotations)
+[!code-csharp[Main](../../../samples/core/Modeling/FluentAPI/ValueGeneratedOnAdd.cs?name=ValueGeneratedOnAdd&highlight=5)]
 
-[!code-csharp[Main](../../../samples/core/Modeling/DataAnnotations/ValueGeneratedOnAdd.cs#Sample)]
+***
 
-> [!WARNING]  
+> [!WARNING]
 > This just lets EF know that values are generated for added entities, it does not guarantee that EF will setup the actual mechanism to generate values. See [Value generated on add](#value-generated-on-add) section for more details.
 
-### Value generated on add or update (Data Annotations)
+### Default values
 
-[!code-csharp[Main](../../../samples/core/Modeling/DataAnnotations/ValueGeneratedOnAddOrUpdate.cs#Sample)]
+On relational databases, a column can be configured with a default value; if a row is inserted without a value for that column, the default value will be used.
```

Comment:
This will likely cause confusion. May be okay to leave for now and fix as part of #1398

Reply:
OK. I plan to do some work on this page as well, specifically to merge the first "conceptual" pass over the value generation patterns, and the second "configuration" pass. I don't think it makes sense to talk about "value generated on add" twice in the same page, once conceptually and once concretely - so I'll try to find a good way to merge them.

We can also look improving the default column guidance as part of #1398, yeah.